### PR TITLE
Switch to JSON mode of cli

### PR DESCRIPTION
**Pull request by nicoangelo**
Hi,

First of all, thank you for this project - definitely a big help for us!

I had to change a few things to actually get it to even start, otherwise it would crash while unlocking. Mostly because the CLI is apparently writing success messages to stderr.

We are using the Bitwarden SaaS offering, so I'm not sure if the BW CLI behaves differently in that case.

Summary:
- Using `--response` switch of the CLI so that it always return JSON responses
- Ignoring stderr, as the JSON message now provides `success`
- Pass around logger to help with debugging 

Let me know what you think.

### CODE REVIEW

**Comment by Lerentis** [comment]
Hey there @nicoangelo ,

Thanks for the kind words and the pr. I will have a closer look at this hopefully tomorrow, but from a first look it makes a lot of sense to me (:

One thing already tho: as the operator makes little sense without the chart I am maintaining the release notes in the artifacthub annotations. Would you mind adding yours there (replacing the current one as it is already released)?

File: src/utils/utils.py [hunk 1]
```diff
@@ -7,44 +7,47 @@ class BitwardenCommandException(Exception):
...
         stderr=subprocess.PIPE,
         close_fds=True,
         shell=True,
         env=system_env)
     out, err = sp.communicate()
-    if err:
-        raise BitwardenCommandException(err)
-    return out.decode(encoding='UTF-8')
+    resp = json.loads(out.decode(encoding='UTF-8'))
+    if os.environ["DEBUG"] != None:
```
Comment (Lerentis) [inline src/utils/utils.py:36]:
```suggestion
    if "DEBUG" in system_env:
```
The `!= None` comparison will try to do the lookup before and result in a key exception

**Review by Lerentis — CHANGES_REQUESTED** [review]
As the `--response` parameter alters the json object by wrapping everything in a `data` object this needs to be changed on every lookup that directly parses the secret json object again. with the current stage the `RegistryCredential` type fails (see `create_dockerlogin` line 22 and 24)

**Comment by Lerentis** [comment]
>We are using the Bitwarden SaaS offering, so I'm not sure if the BW CLI behaves differently in that case.

I think you are on to something here. I am using vaultwarden self hosted for testing. Can you share a anonymised json of some of the cli calls?

File: src/utils/utils.py (hunk 1 above)
Comment (nicoangelo) [reply src/utils/utils.py]: Just shows that python is not my primary language :laughing:
Fixed that and (hopefully) the parts for the `RegistryCredential`

**Comment by nicoangelo** [comment]
Great. Just now realized that you asked me to add the the release notes in the annotations. Will do so in the next few minutes.

**Review by Lerentis — APPROVED** [review]
LGTM :tada:

**Pull request merged**